app/components/settings/settings.component.ts

```diff
@@ -0,0 +1,97 @@
+import { Component, OnDestroy } from "@angular/core";
+import { FormControl } from "@angular/forms";
+import { autobind } from "core-decorators";
+import { Observable, Subscription } from "rxjs";
+
+import { SettingsService } from "app/services";
+import "./settings.scss";
+
+// tslint:disable-next-line:no-var-requires
+const defaultSettings = require("./default-settings.json");
+
+const emptyConfig = "// Place your settings in this file to overwrite the default settings\n{\n\n}";
+
+@Component({
+    selector: "bl-settings",
+    templateUrl: "settings.html",
+})
+export class SettingsComponent implements OnDestroy {
+    public static breadcrumb(params, queryParams) {
+        return { name: "Settings" };
+    }
+
+    public defaultSettingsEditorConfig: CodeMirror.EditorConfiguration = {
+        readOnly: true,
+        lineNumbers: true,
+        mode: "application/json",
+        tabSize: 2,
+        indentUnit: 2,
+    };
+
+    public userSettingsEditorConfig: CodeMirror.EditorConfiguration = {
+        lineNumbers: true,
+        mode: "application/javascript",
+        tabSize: 2,
+        indentUnit: 2,
+        gutters: ["CodeMirror-lint-markers"],
+        lint: true,
+        extraKeys: {
+            "Ctrl-S": this.save,
```

Comment:
yeah ctrl+/ is added for all language but the save depends on the usage of the editor